### PR TITLE
feat(deploy-config-generator): adds new module @sap-ux/deploy-config-generator-shared

### PR DESCRIPTION
**Pull request by cianmSAP**
#2526 

Adds new module @sap-ux/deploy-config-generator-shared 
- Contains base class all deployment generators will extend
- Central location for common error messages and constants used across the deployment generators

### CODE REVIEW

File: packages/deploy-config-generator-shared/src/base/generator.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export abstract class DeploymentGenerator extends Generator {
+    private static _logger: ILogWrapper = DefaultLogger;
+
+    /**
+     * Deployment generator constructor.
+     *
+     * @param args - arguments passed to the generator
+     * @param options - options passed to the generator
+     */
+    constructor(args: string | string[], options: Generator.GeneratorOptions) {
```
Comment (IainSAP) [inline packages/deploy-config-generator-shared/src/base/generator.ts:18]: Just a query. Do we need to define the options supported when launching the deployment config generator?

**Review by IainSAP — APPROVED** [review]
Thanks @cianmSAP 

- Basic cross cutting code to support sub-gens to come
- Tests added to cover

**Review by longieirl — APPROVED** [review]
Thanks @cianmSAP 
Changes look good.

**Review by devinea — APPROVED** [review]
New module set-up looks good. 
changeset ✅

**Pull request merged**